crates/kornia-imgproc/src/features/fast.rs

```diff
@@ -1,162 +1,365 @@
+    pub fn new(
+        image_size: ImageSize,
+        threshold: f32,
+        arc_length: usize,
+        min_distance: usize,
+    ) -> Result<Self, ImageError> {
+        Ok(Self {
+            threshold,
+            min_distance,
+            arc_length,
+            corner_response: Image::from_size_val(image_size, 0.0, CpuAllocator)?,
+            mask: Image::from_size_val(image_size, false, CpuAllocator)?,
+            taken: vec![false; image_size.height * image_size.width],
         })
-        .collect();
+    }
 
-    Ok(keypoints)
-}
+    /// Clears the internal state of the detector, marking it ready to detect again.
+    pub fn clear(&mut self) {
+        self.taken.par_iter_mut().for_each(|px| *px = false);
+    }
+
+    /// Computes the corner response for the input image.
+    ///
+    /// # Arguments
+    ///
+    /// * `src` - The source grayscale image.
+    ///
+    /// # Returns
+    ///
+    /// Returns a reference to the image containing the corner response.
+    pub fn compute_corner_response<A: ImageAllocator>(
+        &mut self,
+        src: &Image<f32, 1, A>,
+    ) -> Result<&Image<f32, 1, CpuAllocator>, ImageError> {
+        if self.corner_response.size() != src.size() {
+            return Err(ImageError::InvalidImageSize(
+                src.width(),
+                src.height(),
+                self.corner_response.width(),
+                self.corner_response.height(),
+            ));
+        }
+
+        let src_slice = src.as_slice();
+
+        let width = src.width();
+        let height = src.height();
+
+        let corner_response = self.corner_response.as_slice_mut();
 
-fn is_fast_corner(
-    src: &[u8],
-    pixel_idx: i32,
-    offsets: [i32; 16],
-    threshold: u8,
-    arc_length: u8,
-) -> bool {
-    let center_pixel = unsafe { *src.get_unchecked(pixel_idx as usize) };
-    let lower_threshold = &center_pixel.saturating_sub(threshold);
-    let upper_threshold = &center_pixel.saturating_add(threshold);
-
-    // Helper to get pixel value efficiently with unchecked access
-    let get_pixel_from_offset =
-        |off_idx: usize| unsafe { src.get_unchecked((pixel_idx + offsets[off_idx]) as usize) };
-
-    // Fast rejection test - check if at least 3 of the 4 high-speed test points are different enough
-    let p1 = get_pixel_from_offset(0);
-    let p5 = get_pixel_from_offset(4);
-    let p9 = get_pixel_from_offset(8);
-    let p13 = get_pixel_from_offset(12);
-
-    let m0 = !(p5 >= lower_threshold || p1 >= lower_threshold && p9 >= lower_threshold);
-    let m1 = !(p5 <= upper_threshold || p1 <= upper_threshold && p9 <= upper_threshold);
-
-    if !m0 && !m1 {
-        return false;
+        const RP: [isize; 16] = [0, 1, 2, 3, 3, 3, 2, 1, 0, -1, -2, -3, -3, -3, -2, -1];
+        const CP: [isize; 16] = [3, 3, 2, 1, 0, -1, -2, -3, -3, -3, -2, -1, 0, 1, 2, 3];
+
+        corner_response[3 * width..(height - 3) * width]
```

Comment:
add comment of what's happening here; i guess we are skipping the +-3 top/down borders (?)